src/MongoDB.EntityFrameworkCore/Storage/MongoClientWrapper.cs

```diff
@@ -85,59 +85,118 @@ public IClientSessionHandle StartSession()
 
     /// <inheritdoc />
     public async Task<IClientSessionHandle> StartSessionAsync(CancellationToken cancellationToken = default)
-        => await _client.StartSessionAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
+        => await _client.StartSessionAsync(null, cancellationToken).ConfigureAwait(false);
 
     /// <inheritdoc />
-    public bool CreateDatabase(IModel model)
+    public bool CreateDatabase(IDesignTimeModel model)
     {
-        var existed = DatabaseExists();
+        var didCreateNewDatabase = !DatabaseExists();
         var existingCollectionNames = _database.ListCollectionNames().ToList();
 
-        foreach (var collectionName in model.GetEntityTypes().Where(e => e.IsDocumentRoot()).Select(e => e.GetCollectionName()))
+        foreach (var entityType in model.Model.GetEntityTypes().Where(e => e.IsDocumentRoot()))
         {
-            if (existingCollectionNames.Contains(collectionName)) continue;
-
-            try
+            var collectionName = entityType.GetCollectionName();
+            if (!existingCollectionNames.Contains(collectionName))
             {
-                _database.CreateCollection(collectionName);
+                try
+                {
+                    _database.CreateCollection(collectionName);
+                    existingCollectionNames.Add(collectionName);
+                }
+                catch (MongoCommandException ex) when (ex.Message.Contains("already exists"))
+                {
+                    // Ignore collection already exists in cases of concurrent creation
+                }
             }
-            catch (MongoCommandException ex) when (ex.Message.Contains("already exists"))
+
+            CreateIndexes(entityType);
+        }
+
+        return didCreateNewDatabase;
+    }
+
+    private void CreateIndexes(IEntityType entityType)
+    {
+        var indexManager = _database.GetCollection<BsonDocument>(entityType.GetCollectionName()).Indexes;
+        var existingIndexNames = indexManager.List().ToList().Select(i => i["name"].AsString).ToList();
+
+        foreach (var index in entityType.GetIndexes())
+        {
+            var name = index.Name ?? DefaultIndexName(index);
+            if (!existingIndexNames.Contains(name))
             {
-                // Ignore collection already exists in cases of concurrent creation
+                indexManager.CreateOne(CreateIndexDocument(index, name));
             }
         }
-
-        return !existed;
     }
 
     /// <inheritdoc />
-    public async Task<bool> CreateDatabaseAsync(IModel model, CancellationToken cancellationToken = default)
+    public async Task<bool> CreateDatabaseAsync(IDesignTimeModel model, CancellationToken cancellationToken = default)
```

Comment:
Why this change?

Reply:
The design-time model contains the indexes but they are removed in the optimized run-time models.

While brekaing, the IMongoClientWrapper interface does state that this interface is for internal use and is not subject to the breaking change policy.